### ISSUE TITLE
[BUG] TypeError is raised if argument is capitalized

### ISSUE THREAD

**Issue by GodSaveTheDoge** [tiangolo/typer#287]
### Describe the bug

If the "entry" function (the one decorated with typer.Typer.command or the one passed to typer.run) expects a capitalized argument, a TypeError will be raised (TypeError: <TheEntryFunction>() got an unexpected keyword argument '<argumentNameLowercase>')

### To Reproduce

* Create a file `main.py` with:

```Python
import typer

app = typer.Typer()

@app.command()
def hello(nAme: str):
    typer.echo(f"Hello {nAme}")


if __name__ == "__main__":
    app()
```

* Call it with:

```bash
python main.py GodSaveTheDoge
```

* It outputs:

```
Traceback (most recent call last):
  File "githubExample.py", line 11, in <module>
    app()
  File "/data/data/com.termux/files/usr/lib/python3.8/site-packages/typer/main.py", line 214, in __call__
    return get_command(self)(*args, **kwargs)
  File "/data/data/com.termux/files/usr/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/data/data/com.termux/files/usr/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/data/data/com.termux/files/usr/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/data/data/com.termux/files/usr/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/data/data/com.termux/files/usr/lib/python3.8/site-packages/typer/main.py", line 497, in wrapper
    return callback(**use_params)  # type: ignore
TypeError: hello() got an unexpected keyword argument 'name'
```

* But I expected it to output:

```
Hello GodSaveTheDoge
```

### Expected behavior

I expected the function to be called with the capitalized version instead.

### Environment

* OS: Android
* Typer Version: 0.3.2
* Python version: 3.8.0

**Comment by captainCapitalism** [comment]
This is actually the `click` case. Snippet bellow will give you the same result. The same will happen when you would like to pass `nAme` as an `click.option`.

```python
import click


@click.command()
@click.argument("nAme")
def hello(nAme):
    print(f"Hello {nAme}")

if __name__ == "__main__":
    hello()

```

Looks like they call `"sOmE_sTrInG".lower()` on both names of commands, the arguments and the options and that creates an issue later.

 The question I have is: why would you like to do that in the first place? I suppose issue the arised when you wanted to create some camel case arguments.

**Comment by GodSaveTheDoge** [comment]
Yep, that was the case... `def main(serviceId)`

**Comment by captainCapitalism** [comment]
You could manipulate arguments passed into your function by doing some decorator. Something like this comes me to mind:

```python
import functools
from typing import List

import typer

app = typer.Typer()


def fix_arguments(to_fix: List[str] = None):
    if to_fix is None:
        to_fix = []

    def some_name(f):
        @functools.wraps(f)
        def wrapper(**kwargs):

            for argument_to_fix in to_fix:
                lower_name = argument_to_fix.lower()
                real_value = kwargs[lower_name]
                kwargs[argument_to_fix] = real_value
                del kwargs[lower_name]

            return f(**kwargs)

        return wrapper
    return some_name


@app.command()
@fix_arguments(["serviceId"])
def hello(
    serviceId: str = typer.Argument(...,),
):
    typer.echo(f"Hello {serviceId}")

if __name__ == "__main__":
    app()
```

Drawback is it does not work with options if you would like them, they will be lowercase `someOption` will become `--someoption`.

I would actually recommend (if you want to break python `snake_case` convention by using `pascalCase`) to break from `pascalCase` itself for a moment.

Treat the command you pass to typer as an adapter and just invoke function inside however you'd like:
```python
import typer

app = typer.Typer()


def some_action_with_service(serviceId):
    typer.echo(serviceId)


@app.command()
def hello(
    service_id: str = typer.Argument(..., ),
):
    some_action_with_service(serviceId=service_id)


if __name__ == "__main__":
    app()
```

**Comment by GodSaveTheDoge** [comment]
I prefer not making stuff more complex than it has to be. I'll just use snake case, it's not really an issue.

**Comment by captainCapitalism** [comment]
I think that would be an ultimate advice. :) 

Was there a reason that made you want to sneak a capital letter into the argument name?

**Comment by GodSaveTheDoge** [comment]
I usually use pascalCase because snake_case is really ugly imho

**Comment by captainCapitalism** [comment]
I understand, I used to think the same but over time I got used to it and it does not matter in the end because there are different conventions in other languages. 

I think the lesson is that the consistency is the key. In Python it is actually expected to use `snake_case` convention. While you will be interfacing with libraries you will have to use snake_case because they use it, unless you will create tons of wrappers to have unified codebase (but that comes with many more issues). Look here for example: https://typer.tiangolo.com/tutorial/subcommands/add-typer/#put-them-together. Once you would have some subcommands, you will need to add some snake_case to code base and the code will have inconsistent style and will be harder to read overall.
